modules/ensemble/lib/widget/slidable.dart

```diff
@@ -136,61 +97,72 @@ class EnsembleSlidableState extends EWidgetState<EnsembleSlidable> {
       enabled: widget.controller.enabled,
       closeOnScroll: widget.controller.closeOnScroll,
       direction: widget.controller.direction ?? Axis.horizontal,
-      dragStartBehavior: widget.controller.dragStartBehavior ?? DragStartBehavior.down,
+      dragStartBehavior:
+          widget.controller.dragStartBehavior ?? DragStartBehavior.down,
       useTextDirection: widget.controller.useTextDirection,
-      startActionPane: _buildActionPane(context, widget.controller.startContent),
-      endActionPane: _buildActionPane(context, widget.controller.endContent),
+      startActionPane: _buildActionPane(context, widget.controller.startDrawer),
+      endActionPane: _buildActionPane(context, widget.controller.endDrawer),
       child: _buildChildWidget(context, widget.controller.child),
     );
   }
 
-  ActionPane? _buildActionPane(BuildContext context, dynamic actionPaneDefinition) {
-    if (actionPaneDefinition == null) return null;
-    
-    ActionPaneOptionsComposite options = ActionPaneOptionsComposite.from(actionPaneDefinition['options']);
-    List<Widget> children = _buildSlidableActions(context, actionPaneDefinition['children']);
+  ActionPane? _buildActionPane(
+      BuildContext context, SlidableDrawerComposite? drawerComposite) {
+    if (drawerComposite == null || drawerComposite.children.isEmpty)
+      return null;
+
+    ActionPaneOptionsComposite options = drawerComposite.options;
+    List<Widget> children =
+        _buildSlidableActions(context, drawerComposite.children);
 
     return ActionPane(
       extentRatio: options.extentRatio,
       motion: options.getMotion(),
-      dragDismissible: options.dragDismissible,
       openThreshold: options.openThreshold,
       closeThreshold: options.closeThreshold,
       children: children,
     );
   }
 
-  List<Widget> _buildSlidableActions(BuildContext context, dynamic actionsDefinition) {
-    if (actionsDefinition == null || actionsDefinition is! List) return [];
+  List<Widget> _buildSlidableActions(
+      BuildContext context, List<dynamic> actionsDefinition) {
+    if (actionsDefinition.isEmpty) return [];
     return actionsDefinition.map<Widget>((action) {
       final IconData? icon = _getIconData(action['icon']);
       final String? label = action['label'];
 
       // Ensure at least one of icon or label is provided
       if (icon == null && (label == null || label.isEmpty)) {
-        return const SizedBox.shrink(); // Return an empty widget if both are missing
+        return const SizedBox
+            .shrink(); // Return an empty widget if both are missing
       }
 
       return SlidableAction(
         onPressed: (context) {
           if (action['onPressed'] != null) {
-            final onPressedAction = EnsembleAction.from(action['onPressed'], initiator: widget);
+            final onPressedAction =
```

Comment:
This certainly can return null which will throw an except below. Check for it and never never ever use force exclamation like "onPressAction!". If you have to do that, something is wrong.

Reply:
yup got the point, thanks, and will take care in future as well, but the same has been implemented at every action, maybe we need to keep updating such mistake while fixing other stuff.